src/net/protocol_buffer.h

```diff
@@ -141,10 +141,7 @@ inline bool
 ProtocolBuffer<tmpl_size>::consume(difference_type v) {
   m_position += v;
 
-  if (remaining())
-    return false;
-
-  return true; 
+  return !static_cast<bool>(remaining());
```

Comment:
This makes it more difficult to read.

Reply:
How about removing the cast and just returning

```
return !remaining();
```

Reply:
No, the style I used originally makes it very clear what is happening at a glance, while you have to think more with the compact style you are suggesting.